obelix/idefix/services/google/client.go

```diff
@@ -0,0 +1,74 @@
+package google
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"google.golang.org/api/calendar/v3"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/structpb"
+	"idefix/devauth"
+	"idefix/operator"
+	"idefix/services/github"
+	"log"
+	"net/http"
+)
+
+// Client represents a minimal GitHub client able to make OAuth2.0 authenticated requests.
+type Client struct {
+	Requester *http.Client
+	Operator  *operator.IdefixOperator
+}
+
+// configure get credentials to connect to GitHub API
+func (c *Client) configure() error {
+	//TODO Credentials API
+	client, err := devauth.New(context.Background(), []string{
+		"https://www.googleapis.com/auth/bigquery",
+		"https://www.googleapis.com/auth/blogger",
+		"https://www.googleapis.com/auth/calendar",
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to configure google client")
+	}
+
+	c.Requester = client.Clt
+
+	return nil
+}
+
+// CalendarNearestEvent check if event's calendar begin before 10 minutes
+func (c *Client) CalendarNearestEvent(taskID string, prm *structpb.Struct) error {
+	var gc GCalendar
+
+	if err := c.configure(); err != nil {
+		return fmt.Errorf("failed to configure github client: %w", err)
+	}
+
+	srv, err := calendar.NewService(context.Background(), option.WithHTTPClient(c.Requester))
+	if err != nil {
+		log.Fatalf("Unable to retrieve Calendar client: %v", err)
+	}
+
+	err = gc.Parse(srv)
+	if err != nil || !errors.Is(err, ErrNoUpcomingEvent) {
+		return fmt.Errorf("failed to parse gcalendar data: %w", err)
+	}
```

Comment:
Can you inline this ?

Reply:
96ecf088b2cf205ee3e11412cc2a3c60073d6498